CHANGELOG.md

```diff
@@ -72,17 +73,25 @@ Adding a new version? You'll need three changes:
   At the same time deprecate `--enable-controller-ingress-extensionsv1beta1` CLI flag.
   [#3710](https://github.com/Kong/kubernetes-ingress-controller/pull/3710)
 
-## [2.9.1]
+## [2.9.2]
 
-> Release date: 2023-03-29
+> Release date: 2023-04-03
 
-### Fixed 
+### Fixed
 
 - Fixed a deadlock in `AdminAPIClientsManager` which could occur when Konnect integration
   was enabled, and multiple `Notify` calls were made in parallel (e.g. when scaling Gateway
   deployment up).
   [#3816](https://github.com/Kong/kubernetes-ingress-controller/pull/3816)
```

Comment:
Let's not cover this here but we might take a consistent approach to either:

- link PRs to `main` (which is being done here)
- link PRs to the relevant release branch 
- link PRs to `main` with a note that it's backported and provide a link to the backported PR as well